include/pqxx/internal/conversions.hxx

```diff
@@ -1156,6 +1156,37 @@ inline constexpr format param_format(std::vector<std::byte, Args...> const &)
 template<typename T> inline constexpr bool is_sql_array<std::vector<T>>{true};
 
 
+#if defined(PQXX_HAVE_SPAN) && __has_include(<span>)
+template<typename T, size_t Extent>
+struct nullness<std::span<T, Extent>> : no_null<std::span<T, Extent>>
+{};
+
+
+template<typename T, size_t Extent>
+struct string_traits<std::span<T, Extent>>
+        : internal::array_string_traits<std::span<T, Extent>>
+{};
+
+
+template<typename T, size_t Extent>
+inline constexpr format param_format(std::span<T, Extent> const &)
+{
+  return format::text;
+}
+
+
+template<size_t Extent>
+inline constexpr format param_format(std::span<std::byte, Extent> const &)
+{
+  return format::binary;
+}
+
+
+template<typename T, size_t Extent>
+inline constexpr bool is_sql_array<std::span<T, Extent>>{true};
```

Comment:
It's an awkward question but...  Is this really what we want for `std::byte`?  In that case, we probably want the SQL equivalent to be `BYTEA`, not an SQL array.

Reply:
you mean:
```lang=c++
template<size_t Extent>
inline constexpr bool is_sql_array<std::span<std::byte, Extent>>{false};
```
?

Reply:
Right.  May need testing though.

Reply:
Oh this is annoying.  My laptop is broken and the borrowed one I'm using right now is too old.  :-(  I was trying to write some test code, especially for the `std::byte` case, but I can't right now.

Reply:
Still waiting for repair on my laptop.  :-/

Reply:
My laptop has been repaired now (and after fixing the firmware/hardware problem they _very helpfully_ wiped my Ubuntu partition and reinstalled Windows over it) so I hope to be able to look into it soon.

Reply:
@fallenworld1 it all took me a lot longer than expected, but there are advantages to a fresh look.

Contiguous ranges of bytes are going to be the norm for handling binary data in libpqxx 8.0 (which requires C++20 or better).  So we'll want to avoid surprises where one range is an SQL array and another is an atomic type.  But that's my problem, not yours.  The really good news here is that there _is_ no conversion support for `std::byte`, so there is no conflict over how to interpret that.

So I'd like to merge your change, with apologies for the delay.